.github/pull_request_template.md

```diff
@@ -22,7 +23,55 @@ Summarize the solution and provide any necessary context needed to understand
 the code change.
 -->
 
+## Follow Up Work
+
+<!--
+Is there anything missing from the solution?
+What still needs to be done?
+-->
+
 ## Related Issues
 <!--
 Please link to any existing GitHub issues pertaining to this PR.
 -->
+
+## Review Guidelines (for Reviewer)
+<!--
+This is a flexible checklist for the reviewer to fill in.
+
+Developers:
+Add extra tasks to the review using list items.
+Skip review tasks using ~~strikethrough~~, or just delete them.
+If you want this pull request to have a specific reviewer, tag them in the list of reviewers.
+
+Reviewer:
+This checklist can help you do your review.
+Add or skip tasks as needed.
+-->
+
+**Does this pull request improve Zebra?**
```

Comment:
I think this should be removed, it reads as confrontational to me. I would expect every PR to improve zebra, otherwise it wouldn't have been written, and focusing on this question in the review checklist could be intimidating for new contributors, as though doubts were being cast upon their ability to improve and contribute to the codebase.

```suggestion
```

Reply:
The goals of a PR and review are not stated, so authors and different reviewers can have conflicting goals - we've had PRs where that's happened.

For example, here is a non-goal:
Have we done everything possible to improve the code? Is it perfect?

And here are some unclear sub-goals:
When is documentation required?
When are tests required?

I'll try to rephrase the goal to be clearer.

Reply:
```suggestion
**Overall Goal: Merged pull requests should improve Zebra.**
<!--
PRs should make incremental improvements to Zebra.
Reviewers can help PR authors write better code.
But PRs don't need to be perfect. It's ok to have follow-up tasks.
-->
```

Reply:
I think this still has those same unintentional connotation issues. How do you feel about this version?

```
- [ ] Description
  - **Does this PR focus on and achieve a clearly defined goal.**
  - The PR description highlights a concrete problem.
  - The solution directly solves the stated problem, rather than circumvents it (as possible).
  - The implemented code is not intended to be thrown away at a later date.
```

Reply:
I'd like to discuss the goals of a PR, because it looks like we have some different goals here.

>  - **Does this PR focus on and achieve a clearly defined goal.**

This is a useful goal, but it overlaps with a later goal:
The PR contains changes to related code (unrelated changes can be split into another PR)

>  - The PR description highlights a concrete problem.

I think we should add this goal to the pull request section:
https://github.com/ZcashFoundation/zebra/pull/1177/files#r513157688

>  - The solution directly solves the stated problem, rather than circumventing it (as possible).

This is a good goal, but I don't know how to check for it.
If we can find a good way to word it, let's add it to the Code section:
https://github.com/ZcashFoundation/zebra/pull/1177/files#diff-b2496e80299b8c3150b1944450bd81c622e04e13d15c411d291db0927d75fd6bR60

>  - The implemented code is not intended to be thrown away at a later date.

I think this is a non-goal: it's ok to add code that we plan to throw away, as long as that's clearly documented using comments and issues. See these related items:
- The PR does everything listed in the tickets that it closes, or the designs that it finishes
- Any follow-up tasks are listed in a GitHub issue
- Any known issues or limitations are documented

I am also trying to keep the template short.

So I'd like to limit the number of sections and checklist items. Let's also look for things to delete - we should delete anything that isn't a top priority. (We can move details to the Zebra book if we need to.)

Reply:
Should we delete this checklist item?

The item is ambiguous - it means different things to different people. So a PR template isn't a good place to discuss the goals of a review.

Let's move this discussion to next week's Zebra sync?

```suggestion
```

Reply:
:+1: